src/react/projects/spark-react/src/SprkStepper/SprkStepper.js

```diff
@@ -0,0 +1,201 @@
+    this.updateSliderPosition = this.updateSliderPosition.bind(this);
+
+    this.state = {
+      activeStepIndex: this.getInitialActiveStep(),
+      sliderStyle: { top: 0, },
+    };
+  }
+
+  handleStepClick(e, indexOfStep) {
+    this.setNewActiveStep(indexOfStep);
+  }
+
+  handleKeyEvents(event) {
+    const keys = {
+      end: 35,
+      home: 36,
+      left: 37,
+      right: 39,
+      up: 38,
+      down: 40,
+    };
+
+    if (event.keyCode === keys.left || event.keyCode === keys.up) {
+      event.preventDefault();
+      this.retreatTab();
+    } else if (event.keyCode === keys.right || event.keyCode === keys.down) {
+      event.preventDefault();
+      this.advanceTab();
+    } else if (event.keyCode === keys.home) {
+      event.preventDefault();
+      this.jumpToFirstTab();
+    } else if (event.keyCode === keys.end) {
+      event.preventDefault();
+      this.jumpToLastTab();
+    }
+  }
+
+  retreatTab() {
+    const { activeStepIndex } = this.state;
+    const { children } = this.props;
+
+    let newIndex = activeStepIndex - 1;
+
+    if (newIndex < 0) {
+      newIndex = children.length - 1;
+    }
+
+    this.setNewActiveStep(newIndex);
+  }
+
+  advanceTab() {
+    const { activeStepIndex } = this.state;
+    const { children } = this.props;
+
+    let newIndex = activeStepIndex + 1;
+
+    if (newIndex >= children.length) {
+      newIndex = 0;
+    }
+
+    this.setNewActiveStep(newIndex);
+  }
+
+  jumpToFirstTab() {
+    this.setNewActiveStep(0);
+  }
+
+  jumpToLastTab() {
+    const { children } = this.props;
+
+    this.setNewActiveStep(children.length - 1);
+  }
+
+  getInitialActiveStep() {
+    const { children } = this.props;
+
+    // the first step is active by default
+    let initialIndex = 0;
+
+    if (children.length > 0) {
+      children.forEach((child, index) => {
+        if (child.props.isSelected) {
+          initialIndex = index;
+        }
+      });
+    }
+
+    return initialIndex;
+  }
+
+  setNewActiveStep(stepIndex){
+    this.setState({ activeStepIndex: stepIndex });
+  }
+
+  updateSliderPosition(position) {
+    this.setState({
+      sliderStyle: { top: position }
+    });
+  }
+
+  render() {
+    const { additionalClasses, children, idString, hasDarkBackground, ...other } = this.props;
+    const { activeStepIndex } = this.state;
+
+    let hasDesc = false;
+    let descCount = 0;
+
+    // count the number of Step children with descriptions. If that doesn't match the total number of children,
+    // warn the user
+    children.forEach((child, index) => {
+      if (child.type.name === SprkStepperStep.name && child.props.children) {
+        descCount++;
+      }
+    });
+
+    if (descCount === children.length) {
+      hasDesc = true;
+    } else if (descCount > 0){
+      console.warn("SprkStepper: If any step has a description, they all must have a description.");
+    }
+
+    let stepper = (
+      <ol
+        className={classnames(
```

Comment:
These dynamic classes should be applied to a div that wraps this `<ol>` or else box and other spacing classes seem to break thing when you try to add some "blue space" around dark background steppers. 

I think the solution here is to have no default padding, add the div, then change the code sample in the docs to pass in a medium box class.